python/hsfs/core/arrow_flight_client.py

```diff
@@ -531,16 +531,7 @@ def create_query_object(self, query, query_str, on_demand_fg_aliases=None):
             fg_connector = _serialize_featuregroup_connector(
                 fg, query, on_demand_fg_aliases
             )
-            if isinstance(fg, feature_group.ExternalFeatureGroup):
-                selected_features = []
-                for feat in query.features:
-                    if feat._feature_group_id == fg.id:
-                        selected_features.append(feat.name)
-                features[fg_name] = [
-                    feat.name for feat in fg.features if feat.name in selected_features
-                ]
-            else:
-                features[fg_name] = [feat.name for feat in fg.features]
+            features[fg_name] = [feat.name for feat in fg.features]
```

Comment:
are the features in `query.features` no longer important? aka. we always use all features?

Reply:
actually this was original behavior, which was correct. It needs all features before registering tables so then it can do select. `feat.name for feat in fg.features` means select all features. 
I am just reverting here.